### PR TITLE
Api/assignments

### PR DESCRIPTION
**Pull request by Padraic-0**
Get and Post submissions

### CODE REVIEW

File: src/api/assignments.ts [hunk 1]
```diff
@@ -4,38 +4,95 @@ import { requireAuthentication } from "../util/authentication.js";
...
+        return isStudent;
+    }
+}), async (req, res) => {
   const assignmentId = req.params.id;
   const submissionData = req.body;
-  // TODO: Implement
-  res.status(201).json({
-    id: 123,
-  });
+  const submission = new Submission(submissionData);
```
Comment (theusaf) [inline src/api/assignments.ts:30]: > The grade, in points, assigned to the student for this sumbission, if one is assigned. Should not be accepted during submission creation, only via update.

File: src/api/assignments.ts [hunk 2]
```diff
@@ -4,38 +4,95 @@ import { requireAuthentication } from "../util/authentication.js";
 import { Course } from "../models/courses.js";
 import { User } from "../models/users.js";
 import { Assignment } from "../models/assignments.js";
+import { Submission } from "../models/submissions.js";
 
 const router = Router();
 
-router.post("/:id/submissions", (req, res) => {
+router.post("/:id/submissions", requireAuthentication({
```
Comment (theusaf) [inline src/api/assignments.ts:11]: This route should also require certain body inputs. It should also save the incoming file to disk or cloud storage.

File: src/api/assignments.ts [hunk 3]
```diff
@@ -4,38 +4,95 @@ import { requireAuthentication } from "../util/authentication.js";
...
-      },
-    ],
+  const assignment = await Assignment.findById(assignmentId);
+  const submissions = await Submission.findAll({
+    where: {
+        assignmentId: assignment.id
+    },});
+
+  res.status(200).json({
+    submissions: submissions,
```
Comment (theusaf) [inline src/api/assignments.ts:66]: Be sure to send back data using the `.toJSON()` to avoid potential issues

Comment (theusaf) [reply src/api/assignments.ts:66]: In addition, the response should contain URLs to the files in the submissions

**Review by theusaf — CHANGES_REQUESTED** [review]

File: src/models/submissions.ts
Comment (theusaf) [inline src/models/submissions.ts:1]: Considering the data saved is a buffer, the return of the `toJSON` should return a URL instead. Also, the `save` method should be overridden to save the buffer to the storage bucket / firestore.
To do this, consider adding a `fileURL` or `fileId` to be able to access  the file data at a later point. (Or design the save system to use the id of the submission).

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

File: src/models/submissions.ts
Comment (Padraic-0) [reply src/models/submissions.ts:1]: Im not sure what that means

**Review by theusaf — APPROVED** [review]

**Pull request merged**